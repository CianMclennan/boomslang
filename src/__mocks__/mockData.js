export const AccordianData = [
	{
		title: {
			component: 'Text',
			text: 'title1',
		},
		content: {
			component: 'Text',
			text: 'Nibh inceptos amet laoreet lobortis facilisi sapien sit varius interdum',
		},
	},
	{
		title: 'title2',
		content:
			'PraesentAliquam turpis viverra sem aenean neque orci amet dolor porta',
	},
];

export default {
	screen_id: 'mock_screen_id',
	component: 'Layout',
	child: {
		component: 'Accordian',
		data: AccordianData,
	},
};
