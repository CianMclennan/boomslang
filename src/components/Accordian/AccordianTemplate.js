export default {
	title: {
		component: 'Text',
		text: 'test',
	},
	content: {
		component: 'Placeholder',
	},
};
