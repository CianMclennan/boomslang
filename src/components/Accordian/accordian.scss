.accordian {
	min-height: 6.25rem;

	&__header {
		background-color: sandybrown;
		padding: 0.3rem;
		font-size: 1.2rem;
	}

	&__content {
		background-color: pink;
		min-height: 12.5rem;

		&--hidden {
			display: none;
		}
	}
}
