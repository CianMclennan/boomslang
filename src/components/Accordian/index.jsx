import React from 'react';
import styled from 'styled-components';

const Accordian = styled.div`
    min-height: 100px;
    & .title-bar {
        background-color: sandybrown;
        padding: 5px;
        font-size: 1.2em;
    }
    & .content {
        background-color: pink;
    }
`;


export default (props) => {
    const content = (props.data || []).map((section, index) => {
        return <div key={index} className="section">
            <div className="title-bar">
                {section.title}
            </div>
            <div className="content">
                {section.content}
            </div>
        </div>
    });

    return <Accordian>
        {content}
    </Accordian>
}