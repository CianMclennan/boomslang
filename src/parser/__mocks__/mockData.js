export default {
	component: 'Layout',
	child: {
		component: 'Accordian',
		data: [
			{
				title: {
					component: 'Text',
					text: 'title1',
				},
				content: {
					component: 'Text',
					text:
						'Nibh inceptos amet laoreet lobortis facilisi sapien sit varius interdum',
				},
			},
			{
				title: 'title2',
				content:
					'PraesentAliquam turpis viverra sem aenean neque orci amet dolor porta',
			},
			{
				title: 'title3',
				content:
					'Sollicitudin scelerisque nisl eu laoreet torquent accumsan proin urna tortor',
			},
			{
				title: 'title4',
				content:
					'Quam nam tortor lorem odio sed posuere pharetra volutpat feugiat',
			},
			{
				title: 'title5',
				content:
					'Nascetur torquent ligula vel aliquam habitant dignissim malesuada nullam accumsan',
			},
		],
	},
};
