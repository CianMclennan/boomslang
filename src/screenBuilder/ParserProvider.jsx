import { createContext, useContext } from 'react';

const ParserContext = createContext();

export const useParser = () => {
	return useContext(ParserContext);
};

export default ParserContext.Provider;
