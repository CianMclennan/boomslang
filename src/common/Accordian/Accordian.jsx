import React, { useState } from 'react';
import { ACCORDIAN, HEADER, CONTENT, CONTENT_HIDDEN } from './constants'
import './accordian.scss'

export default (props) => {
    const [selected, setSelected] = useState(0);    

    const handleHeaderClick = (index) => {
        const newIndex = index === selected ? -1 : index;
        setSelected(newIndex);
    }
    const content = (props.data || []).map(({title, content}, index) => {
        const isHidden = index !== selected;
        return <div key={index}>
            <div className={HEADER} onClick={()=>handleHeaderClick(index)}>
                {title}
            </div>
            <div className={`${CONTENT} ${isHidden && CONTENT_HIDDEN}`}>
                {content}
            </div>
        </div>
    });

    return <div className={ACCORDIAN}>
        {content}
    </div>
}