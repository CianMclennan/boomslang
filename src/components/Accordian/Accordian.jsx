import './accordian.scss';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { useParser } from 'src/screenBuilder/ParserProvider.jsx';
import React, { useState } from 'react';

const Accordian = ({ data, path }) => {
	const [selected, setSelected] = useState(0);
	const parse = useParser();
	const handleHeaderClick = (index) => {
		const newIndex = index === selected ? -1 : index;
		setSelected(newIndex);
	};

	const content = (data || []).map(({ title, content }, index) => {
		const isHidden = index !== selected;
		const contentCN = classnames('accordian__content', {
			'accordian__content--hidden': isHidden,
		});
		return (
			<section key={index}>
				<div
					className="accordian__header"
					onClick={() => handleHeaderClick(index)}
				>
					{parse({ ...title, path: `${path}/data/${index}/title` })}
				</div>
				<div className={contentCN}>
					{parse({ ...content, path: `${path}/data/${index}/content` })}
				</div>
			</section>
		);
	});

	return <div className="accordian">{content}</div>;
};

Accordian.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			title: PropTypes.any.isRequired,
			content: PropTypes.any.isRequired,
		})
	),
	path: PropTypes.string,
};

export default Accordian;
