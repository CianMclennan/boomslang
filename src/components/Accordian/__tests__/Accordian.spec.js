import Accordian from '../Accordian.jsx';
import { AccordianData } from 'src/__mocks__/mockData.js';
import ParserProvider from 'src/screenBuilder/ParserProvider.jsx';
import defaultParser from 'src/screenBuilder/defaultParser.js';
import React, { Suspense } from 'react';
import { fireEvent, render } from '@testing-library/react';

const renderAccodian = (data, deleteHandler) =>
	render(
		<div data-testid="test-id">
			<ParserProvider value={defaultParser}>
				<Suspense fallback={<></>}>
					<Accordian data={data} deleteHandler={deleteHandler} />
				</Suspense>
			</ParserProvider>
		</div>
	);

describe('Accordian Component:', () => {
	it('should render', () => {
		const { container } = renderAccodian(AccordianData);
		expect(container).toMatchSnapshot();
	});
	it('should display delete button when passed a "deleteHandler"', () => {
		let deletedIndex = null;
		const deleteHandler = jest.fn((e, i) => {
			deletedIndex = i;
		});
		const { container, getByTestId } = renderAccodian(
			AccordianData,
			deleteHandler
		);
		const deleteBtn = getByTestId('test-id').querySelector(
			'.editor-icon-btn__delete'
		);
		expect(container).toMatchSnapshot();
		fireEvent.click(deleteBtn);
		expect(deletedIndex).toBe(0);
		expect(deleteHandler).toBeCalledTimes(1);
	});
});
