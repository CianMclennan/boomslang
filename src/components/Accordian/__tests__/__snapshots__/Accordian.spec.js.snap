// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component: should display delete button when passed a "deleteHandler" 1`] = `
<div>
  <div
    data-testid="test-id"
  >
    <div
      class="accordian"
    >
      <section>
        <div
          class="accordian__header"
        >
          <span
            style="font-size: 2rem;"
          >
            title1
          </span>
          <button
            class="editor-icon-btn editor-icon-btn__delete"
          />
        </div>
        <div
          class="accordian__content"
        >
          <span>
            Nibh inceptos amet laoreet lobortis facilisi sapien sit varius interdum
          </span>
        </div>
      </section>
      <section>
        <div
          class="accordian__header"
        >
          title2
          <button
            class="editor-icon-btn editor-icon-btn__delete"
          />
        </div>
        <div
          class="accordian__content accordian__content--hidden"
        >
          PraesentAliquam turpis viverra sem aenean neque orci amet dolor porta
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`Accordian Component: should render 1`] = `
<div>
  <div
    data-testid="test-id"
  >
    <div
      class="accordian"
      style="display: none;"
    >
      <section>
        <div
          class="accordian__header"
        />
        <div
          class="accordian__content"
        />
      </section>
      <section>
        <div
          class="accordian__header"
        >
          title2
        </div>
        <div
          class="accordian__content accordian__content--hidden"
        >
          PraesentAliquam turpis viverra sem aenean neque orci amet dolor porta
        </div>
      </section>
    </div>
  </div>
</div>
`;
